core/libp2p/network/connection_manager.hpp

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CONNECTION_MANAGER_HPP
+#define KAGOME_CONNECTION_MANAGER_HPP
+
+#include <memory>
+
+#include "libp2p/basic/garbage_collectable.hpp"
+#include "libp2p/connection/capable_connection.hpp"
+
+namespace libp2p::network {
+
+  // TODO(warchant): when connection is closed ('onDisconnected' event fired),
+  // manager should remove it from storage PRE-212
+
+  struct ConnectionManager : public basic::GarbageCollectable {
+    using Connection = connection::CapableConnection;
+    using ConnectionSPtr = std::shared_ptr<Connection>;
+
+    enum class Connectedness {
+      NOT_CONNECTED,  ///< we don't know peer's addresses, and are not connected
+      CONNECTED,      ///< we have at least one connection to this peer
+      CAN_CONNECT,  ///< we know address of this peer, and we can dial to create
+                    ///< a connection
+      CAN_NOT_CONNECT  ///< we know address of this peer, but can not dial (no
+      ///< transports supported)
+    };
+
+    ~ConnectionManager() override = default;
+
+    // get list of all connections (including inbound and outbound)
+    virtual std::vector<ConnectionSPtr> getConnections() const = 0;
+
+    // get list of all outbound connections to a given peer.
+    virtual std::vector<ConnectionSPtr> getConnectionsToPeer(
+        const peer::PeerId &p) const = 0;
+
+    // get best connection to a given peer
```

Comment:
Are we going to use different strategies for choosing the best connection? If yes, some kind of callable of something like that should be passed to that method, which also may be a default one in .hpp implementation

Reply:
Ah, I see a ToDo below. I think, if we are going to implement it, it should be here already, by design, for example, like a default parameter